cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -27,19 +32,139 @@
  */
 public class BasicWorkflowToken implements WorkflowToken {
   private Map<String, Map<String, Long>> mapReduceCounters;
+  private Map<String, List<NodeValueEntry>> tokenValueMap = Maps.newHashMap();
```

Comment:
Made the `tokenValueMap` final. 
Can you please explain the comment 
 >> I think it's better for the actual instance of the List be immutable.
Since we keep adding the `NodeValueEntry` to the list if the `put` on the same key is made by subsequent nodes, not sure how making the List immutable will work.